modules/frontend/traceid_handlers.go

```diff
@@ -0,0 +1,91 @@
+package frontend
+
+import (
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level" //nolint:all //deprecated
+	"github.com/grafana/dskit/user"
+	"github.com/grafana/tempo/modules/frontend/combiner"
+	"github.com/grafana/tempo/modules/frontend/pipeline"
+	"github.com/grafana/tempo/modules/overrides"
+	"github.com/grafana/tempo/pkg/api"
+)
+
+// newTraceIDHandler creates a http.handler for trace by id requests
+//
+// todo: the trace by id path consumes a lot of resources for large traces b/c it
+// repeatedly unmarshals/marshals the trace data. Note that it occurs once here
+// to marshal into the proto format and again in the deduper middleware. we should
+// collapse this into the combiner where the data is already unmarshalled
```

Comment:
I don't think this comment is up-to-date anymore? This function doesn't marshal/unmarshal anymore and the deduper directly accepts a `*tempopb.Trace` now.

Reply:
good call. removed.